chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -974,6 +974,12 @@ where
                             );
                         });
                 }
+            } else {
+                warn!(
+                    logger1,
+                    "Failed to fetch block hash for block number";
+                    "number" => block_number
```

Comment:
I am not really deep enough in the logic of all this to review it, but each log message that's related to a specific provider should include an indication of which provider was used. Does `logger1` already contain that? Otherwise, it should be added either to the logger or to this log message.